javascript/ql/src/semmle/javascript/security/dataflow/TaintedPath.qll

```diff
@@ -67,6 +67,40 @@ module TaintedPath {
         read.getPropertyName() != "length" and
         srclabel = dstlabel
       )
+      or
+      // string method calls of interest
+      exists(DataFlow::MethodCallNode mcn | srclabel = dstlabel |
+        exists(string substringMethodName |
+          substringMethodName = "substr" or
+          substringMethodName = "substring" or
+          substringMethodName = "slice"
+        |
+          mcn.calls(src, substringMethodName) and
+          // to avoid very dynamic transformations, require at least one fixed index
+          exists(mcn.getAnArgument().asExpr().getIntValue()) and
+          dst = mcn
+        ) or
+        exists(string argumentlessMethodName |
+          argumentlessMethodName = "toLocaleLowerCase" or
+          argumentlessMethodName = "toLocaleUpperCase" or
+          argumentlessMethodName = "toLowerCase" or
+          argumentlessMethodName = "toUpperCase" or
+          argumentlessMethodName = "trim" or
+          argumentlessMethodName = "trimLeft" or
+          argumentlessMethodName = "trimRight"
+        |
+          mcn.calls(src, argumentlessMethodName) and
+          dst = mcn
+        )
+        or
+        exists(DataFlow::PropRead fixedRead |
+          mcn.calls(src, "split") and
+          // to avoid very dynamic transformations, propagate into a fixed index of the array
+          dst = fixedRead and
+          fixedRead = mcn.getAPropertyRead() and
+          exists(fixedRead.getPropertyNameExpr().getIntValue())
```

Comment:
The question of what label to use here becomes a bit easier if we instead require that the split delimiter is not a slash.